File: simple-chatbot/go/history.go
package main

import (
	"sync"

	"github.com/firebase/genkit/go/ai"
)

var (
	history = make(map[string][]*ai.Message)
	mu      sync.RWMutex
)

func LoadHistory(sessionID string) []*ai.Message {
	mu.RLock()
	defer mu.RUnlock()
	return history[sessionID]
}

func SaveHistory(sessionID string, messages []*ai.Message) {
	mu.Lock()
	defer mu.Unlock()
	history[sessionID] = messages
}

File: simple-chatbot/src/history.ts
import { MessageData } from 'genkit/beta';

const history: Record<string, MessageData[]> = {};

export async function loadHistory(
  sessionId: string
): Promise<MessageData[] | null> {
  return history[sessionId] || null;
}

export async function saveHistory(
  sessionId: string,
  messages: MessageData[]
): Promise<void> {
  history[sessionId] = messages;
}
